downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/GenerateGOAnnotationFile/BiologicalProcessAnnotationBuilder.java

```diff
@@ -0,0 +1,122 @@
+package org.reactome.release.downloadDirectory.GenerateGOAnnotationFile;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.schema.SchemaClass;
+
+import java.util.*;
+
+public class BiologicalProcessAnnotationBuilder {
+
+    private static final int MAX_RECURSION_LEVEL = 2;
```

Comment:
A comment about how we only look at the parent and "grandparent" pathways on a reaction to explain why the value is 2 would be great here.